controllers/ajax_controller.py

```diff
@@ -14,10 +15,61 @@
 from models.account import Account
 from models.event import Event
 from models.favorite import Favorite
+from models.mobile_client import MobileClient
 from models.sitevar import Sitevar
 from models.typeahead_entry import TypeaheadEntry
 
 
+class AccountInfoHandler(LoggedInHandler):
+    """
+    For getting account info.
+    Only provides logged in status for now.
+    """
+    def get(self):
+        self.response.headers['content-type'] = 'application/json; charset="utf-8"'
+        user = self.user_bundle.user
+        self.response.out.write(json.dumps({
+            'logged_in': True if user else False,
+            'user_id': user.user_id() if user else None
+        }))
+
+
+class AccountRegisterFCMToken(LoggedInHandler):
+    """
+    For adding/updating an FCM token
+    """
+    def post(self):
+        if not self.user_bundle.user:
+            self.response.set_status(401)
+            return
+
+        user_id = self.user_bundle.user.user_id()
+        fcm_token = self.request.get('fcm_token')
+        uuid = self.request.get('uuid')
+        display_name = self.request.get('display_name')
+        client_type = ClientType.WEB
+
+        query = MobileClient.query(
+                MobileClient.user_id == user_id,
+                MobileClient.device_uuid == uuid,
+                MobileClient.client_type == client_type)
+        if query.count() == 0:
+            # Record doesn't exist yet, so add it
+            MobileClient(
+                parent=ndb.Key(Account, user_id),
+                user_id=user_id,
+                messaging_id=fcm_token,
+                client_type=client_type,
+                device_uuid=uuid,
+                display_name=display_name).put()
+        else:
+            # Record already exists, update it
+            client = query.fetch(1)[0]
```

Comment:
Should we also update the user id here? This lets us account for the different users on one device, since tokens are tied to the browser and not the account? This way, we just keep the browser tied to the last user who logged in on it.